python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py

```diff
@@ -41,24 +48,78 @@ class LightGBMTrainer(Trainer):
             dataset. All datasets will be transformed by the ``preprocessor`` if
             one is provided. All non-training datasets will be used as separate
             validation sets, each reporting a separate metric.
-        lightgbm_config: LightGBM training parameters. Refer to
-            `LightGBM documentation <https://lightgbm.readthedocs.io/en\
-/latest/Parameters.html>`_ for a list of possible parameters.
+        lightgbm_config: LightGBM training parameters passed to ``lightgbm.train()``.
+            Refer to `LightGBM documentation <https://lightgbm.readthedocs.io>`_
+            for a list of possible parameters. Must contain a "params" key.
+            You can also pass a "dmatrix_params" key, which should be a dict
+            of ``dataset name:dict of kwargs`` passed to respective
+            :class:`xgboost_ray.RayDMatrix` initializations.
         scaling_config: Configuration for how to scale data parallel training.
         run_config: Configuration for the execution of the training run.
         preprocessor: A ray.ml.preprocessor.Preprocessor to preprocess the
             provided datasets.
         resume_from_checkpoint: A checkpoint to resume training from.
     """
 
+    _use_lightgbm: bool = True
+
     def __init__(
         self,
         label_column: str,
         datasets: Dict[str, GenDataset],
-        lightgbm_config: Optional[Dict[str, Any]] = None,
+        lightgbm_config: Dict[str, Any],
         scaling_config: Optional[ScalingConfig] = None,
         run_config: Optional[RunConfig] = None,
         preprocessor: Optional[Preprocessor] = None,
         resume_from_checkpoint: Optional[Checkpoint] = None,
     ):
-        raise NotImplementedError
+        if TRAIN_DATASET_KEY not in datasets:
+            raise KeyError(
+                f"'{TRAIN_DATASET_KEY}' key must be preset in `datasets`. "
+                f"Got {list(datasets.keys())}"
+            )
+        if PARAMS_KEY not in lightgbm_config:
+            raise KeyError(
+                f"`lightgbm_config` must contain a '{PARAMS_KEY}' key. "
+                f"Got {list(lightgbm_config.keys())}"
+            )
+        self._validate_dmatrix_params(lightgbm_config, datasets)
+        self.label_column = label_column
+        self.lightgbm_config = lightgbm_config
+        super().__init__(
+            datasets=datasets,
+            scaling_config=scaling_config,
+            run_config=run_config,
+            preprocessor=preprocessor,
+            resume_from_checkpoint=resume_from_checkpoint,
+        )
+
+    def training_loop(self) -> None:
+        lightgbm_config = self.lightgbm_config.copy()
```

Comment:
Copying may not be necessary here? This is not going to affect the Trainer that's on the driver, which is the one that the user is going to interact with. 

But I'm also fine keeping as is, this is just a minor point.

Reply:
I'd leave this in just to be sure